### PR TITLE
some design fixes

### PR DESCRIPTION
**Pull request by Areyana**
From the begining we used external screenshot maker for DocLoc Test Cases, after some researching i implemented full window screenshots and made it work for DocLoc cases too (in a combined maker), but it's not including system dialog cases, so i implemented new DocLocScreenshotMaker, that will use External maker for "normal" screenshots like it was and Internal maker for full window screenshots to make it work.

### CODE REVIEW

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt [hunk 1]
```diff
@@ -146,12 +146,12 @@ abstract class DocLocScreenshotTestCase(
                 resourcesDirsProvider,
                 resourceFileNamesProvider
             ),
-            screenshotMaker = CombinedScreenshotMaker(
-                preferredScreenshotMaker = InternalScreenshotMaker(kaspresso.device.activities, screenshotParams),
-                fallbackScreenshotMaker = ExternalScreenshotMaker(
+            screenshotMaker = DocLocScreenshotMaker(
+                screenshotMaker = ExternalScreenshotMaker(
```
Comment (VladislavSumin) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt:150]: before changes we priority use `InternalScreenshotMaker` for all screenshot, why we change default screenshots to external for docLocTest?

Comment (VladislavSumin) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt:150]: Why we don`t changed that for com/kaspersky/kaspresso/kaspresso/Kaspresso.kt:740?

**Comment by matzuk** [comment]
@Areyana please add the purpose of this PR

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt [hunk 2]
```diff
@@ -146,12 +146,12 @@ abstract class DocLocScreenshotTestCase(
                 resourcesDirsProvider,
                 resourceFileNamesProvider
             ),
-            screenshotMaker = CombinedScreenshotMaker(
-                preferredScreenshotMaker = InternalScreenshotMaker(kaspresso.device.activities, screenshotParams),
-                fallbackScreenshotMaker = ExternalScreenshotMaker(
+            screenshotMaker = DocLocScreenshotMaker(
```
Comment (matzuk) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt:149]: why?

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.kaspersky.kaspresso.device.screenshots.screenshotmaker
+
+import java.io.File
+
+class DocLocScreenshotMaker(
```
Comment (matzuk) [inline kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt:5]: the purpose of the class?

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt (hunk 1 above)
Comment (Areyana) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt:150]: We used External Screenshots only for Doc Loc, for Full window screenshots we'll use internal channel, as for Kaspresso:740 i'll fix it

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt (hunk 2 above)
Comment (Areyana) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt:149]: From the begining we used external screenshot maker for DocLoc Test Cases, after some researching i implemented full window screenshots and made it work for DocLoc cases too (in a combined maker), but it's not including system dialog cases, so i implemented new DocLocScreenshotMaker, that will use External maker for "normal" screenshots like it was and Internal maker for full window screenshots to make it work.

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt (hunk 3 above)
Comment (Areyana) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt:5]: Same as above

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt (hunk 1 above)
Comment (Areyana) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt:150]: @VladislavSumin  for what do we need to change that in Kaspresso:740 ? That's screenshots only for base tests, and it was originally Internal and fallback for external

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt (hunk 3 above)
Comment (VladislavSumin) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt:5]: please add code comments with description why we need this class

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt (hunk 1 above)
Comment (VladislavSumin) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/testcases/api/testcase/DocLocScreenshotTestCase.kt:150]: Ok, I further discussed this questions with @AzamatCherchesov and we came to the conclusion that this decision is justified

File: kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt (hunk 3 above)
Comment (Areyana) [reply kaspresso/src/main/kotlin/com/kaspersky/kaspresso/device/screenshots/screenshotmaker/DocLocScreenshotMaker.kt:8]: done

**Review by AzamatCherchesov — APPROVED** [review]

**Pull request merged**